File: 277_find_the_celebrity.java
/* The knows API is defined in the parent class Relation.
      boolean knows(int a, int b); */

public class Solution extends Relation {
    public int findCelebrity(int n) {
        int celebrity = 0;
        for (int i = 0; i < n; i++) if (knows(celebrity, i)) celebrity = i;
        for (int j = 0; j < n; j++) {
            if (j == celebrity) continue;

            // if j does not know candidate or candidaet knows j, return -1
            if ( !knows(j, celebrity) || knows(celebrity, j)) return -1;
        }
        return celebrity;
    }
}

File: python/277_find_the_celebrity.py
# The knows API is already defined for you.
# @param a, person a
# @param b, person b
# @return a boolean, whether a knows b
# def knows(a, b):

class Solution(object):
    def findCelebrity(self, n):
        """
        :type n: int
        :rtype: int
        """
        celebrity = 0
        for i in range(1, n):
            if knows(celebrity, i):
                celebrity = i

        for i in range(n):
            if celebrity != i:
                if knows(celebrity, i):
                    return -1
                if not knows(i, celebrity):
                    return -1

        return celebrity

